core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+    });
+
+    await this.clickOn(languageUpdateBar);
+    await this.clickOn(language);
+  }
+
+  /**
+   *  This function verifies that the selected language is displayed correctly.
+   */
+  async expectSelectedLanguageToBe(expectedLanguage: string): Promise<void> {
+    await this.page.waitForSelector(languageUpdateBar);
+    const languageDropdown = await this.page.$(languageUpdateBar);
+    if (!languageDropdown) {
+      throw new Error('Category dropdown not found.');
+    }
+    await languageDropdown.click();
+
+    const selectedLanguage = await this.page.evaluate(() => {
+      const matOption = document.querySelector(
+        '.mat-option.mat-selected'
+      ) as HTMLElement;
+      if (!matOption) {
+        throw new Error('Selected language option not found.');
+      }
+      const selectedLanguageText = matOption.innerText.trim();
+
+      /**
+       * Extract words within parentheses using regex العربية (Arabic) -> Arabic
+       */
+      const matches = selectedLanguageText.match(/\(([^)]+)\)/);
+      if (matches && matches.length >= 2) {
+        return matches[1].trim();
+      } else {
+        // If no parentheses found, assume the whole text is the language.
+        // For eg : English.
+        return selectedLanguageText.trim();
+      }
+    });
+
+    if (selectedLanguage === expectedLanguage) {
+      showMessage(
+        `The language ${selectedLanguage}` + ' is same as expectedLanguage.'
+      );
+    } else {
+      throw new Error('Language is not correct.');
+    }
+    await this.page.keyboard.press('Enter');
+  }
+
+  /**
+   * This function helps in adding tags.
+   */
+  async addTags(TagNames: string[]): Promise<void> {
+    for (let i = 0; i < TagNames.length; i++) {
+      await this.clickOn(addTags);
+      await this.type(addTags, TagNames[i].toLowerCase());
+      await this.page.keyboard.press('Tab');
+    }
+  }
+
+  /**
+   * This function checks if tags are successfully added.
+   */
+  async expectTagsToBeAdded(expectedTags: string[]): Promise<void> {
+    const lowercaseExpectedTags = expectedTags.map(tag => tag.toLowerCase());
+    await this.page.waitForSelector('mat-chip-list');
+
+    const addedTags = await this.page.evaluate(() => {
+      const tagElements = Array.from(document.querySelectorAll('mat-chip'));
+      return tagElements
+        .map(tag => {
+          const textContent =
+            tag.querySelector('.mat-chip-remove')?.previousSibling?.textContent;
+          return textContent ? textContent.trim() : '';
+        })
+        .filter(Boolean);
+    });
+
+    for (const expectedTag of lowercaseExpectedTags) {
+      if (!addedTags.includes(expectedTag)) {
+        throw new Error(`Tag "${expectedTag}" was not added.`);
+      }
+    }
+
+    showMessage('All expected tags were added successfully.');
+  }
+
+  /**
+   * This function allows you to preview the exploration.
+   */
+  async previewSummary(): Promise<void> {
+    await this.page.waitForSelector(
```

Comment:
See above re waiting for the button to be enabled in a general way.

Reply:
In each utility file, the implementation of waitFor is similiar for eg.
In `blog-post-editor-utils.ts` -> 
-  await this.page.waitForSelector('button.e2e-test-confirm-button');
-   await this.page.waitForSelector(blogDashboardAuthorDetailsModal);
- await this.page.waitForSelector(
      'button.e2e-test-save-author-details-button:not([disabled])'
    );
etc

In `logged-in-users-utils.ts` ->
-     await this.page.waitForXPath('//a[contains(text(),"create one here")]');
 and so on....

Thanks